src/onegov/org/views/ticket.py

```diff
@@ -1069,6 +1074,72 @@ def action_link(ticket):
     }
 
 
+@OrgApp.html(model=ArchivedTicketsCollection, name='delete',
+             request_method='DELETE', permission=Secret)
+def view_delete_all_archived_tickets(self, request):
+    tickets = self.query().filter_by(state='archived')
+
+    errors, ok = delete_tickets_and_related_data(request, tickets)
+    if errors:
+        msg = request.translate(_(
+            "${success_count} tickets deleted, "
+            "${error_count} are not deletable",
+            mapping={'success_count': len(ok), 'error_count': len(errors)},
+        ))
+        request.message(msg, 'warning')
+    else:
+        msg = request.translate(_(
+            "${success_count} tickets deleted.",
+            mapping={'success_count': len(ok)}
+        ))
+        request.message(msg, 'success')
+
+
+def delete_tickets_and_related_data(
+    request: 'CoreRequest', tickets: 'Query[Ticket]'
+) -> tuple[list['Ticket'], list['Ticket']]:
+
+    not_deletable, successfully_deleted = [], []
+
+    for ticket in tickets:
+        if (hasattr(
+                ticket.handler, 'ticket_deletable')
+                and not ticket.handler.ticket_deletable):
+            not_deletable.append(ticket)
+
+            ticket.redact_data()
+            continue
+
+        delete_messages_from_ticket(request, ticket.number)
+
+        if hasattr(ticket.handler, 'prepare_delete_ticket'):
+            ticket.handler.prepare_delete_ticket()
+
+        delete_files_and_submissions_from_ticket(request, ticket)
+
+        request.session.delete(ticket)
+        successfully_deleted.append(ticket)
+
+    return not_deletable, successfully_deleted
+
+
+def delete_messages_from_ticket(request: 'CoreRequest', number: str):
+    messages = MessageCollection(
+        request.session, channel_id=number
+    )
+    for message in messages.query():
+        messages.delete(message)
+
+
+def delete_files_and_submissions_from_ticket(request: 'CoreRequest',
+                                             ticket: 'Ticket'):
+    submission = getattr(ticket.handler, 'submission', None)
+    if submission:
+        for file in submission.files:
+            request.session.delete(file)
```

Comment:
```suggestion
```
I think you can get rid of this, `cascade` should take care of this for you.

Reply:
Probably yes, I'll have a test for that.